src/main/java/org/folio/circulation/services/agedtolost/ChargeLostFeesWhenAgedToLostService.java

```diff
@@ -125,8 +125,14 @@ private CompletableFuture<Result<Void>> chargeLostFeesForLoans(
     Result<List<LoanToChargeFees>> loansToChargeFeesResult) {
 
     return loansToChargeFeesResult
-      .after(loansToChargeFees -> allOf(loansToChargeFees, this::chargeLostFees))
-      .thenApply(r -> r.map(ignored -> null));
+      .after(loans -> allOf(loans, this::processLoan))
+      .thenApply(Result::mapEmpty);
+  }
+
+  private CompletableFuture<Result<Void>> processLoan(LoanToChargeFees loan) {
+    return loan.shouldCloseLoanWhenActualCostUsed()
+      ? updateBillingInfoAndCloseLoanAsLostAndPaid(loan)
+      : chargeLostFees(loan);
```

Comment:
Can we flip the condition?
```suggestion
    return loan.shouldChargeLostItemFess()
      ? chargeLostFees(loan)
      : updateBillingInfoAndCloseLoanAsLostAndPaid(loan);
```